ui/packages/app/src/components/Dataset/ExampleComponents/Dataframe.svelte

```diff
@@ -1,23 +1,69 @@
 <script lang="ts">
-	export let value: Array<Array<string | number>>;
+	//@ts-ignore
+	import { csvParseRows, tsvParseRows } from "d3-dsv";
+	export let value: Array<Array<string | number>> | string;
+	export let samples_dir: string;
+	let hovered = false;
+	let loaded = Array.isArray(value);
+
+	$: if (!loaded && typeof value === "string" && /\.[a-zA-Z]+$/.test(value)) {
+		fetch(samples_dir + value)
+			.then((v) => v.text())
+			.then((v) => {
+				try {
+					if ((value as string).endsWith("csv")) {
+						value = csvParseRows(v);
```

Comment:
Is this reading in the whole file? If so, I don't think we need to - just at most the first 5 rows and columns rights?

Reply:
Yeah, that’s true. I guess some CSVs could be pretty huge.